File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int n) {
    unsigned int i;
    unsigned int sum_squares = 0;
    
    for (i = 1; i <= n; i++ ) {
        sum_squares += i * i;
    }

    return sum_squares;
}

unsigned int square_of_sum(unsigned int n) {
    unsigned int i;
    unsigned int sum = 0;

    for (i = 1; i <= n; i++ ) {
        sum += i;
    }
    sum = sum * sum;
   
    return sum;
}

unsigned int difference_of_squares(unsigned int n) {
    return square_of_sum(n) - sum_of_squares(n);
}

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    square_sum = 0
    for item in range(1, number + 1):
        square_sum += item
    return square_sum ** 2


def sum_of_squares(number):
    sum_square = 0
    for item in range(1, number + 1):
        sum_square += item ** 2
    return sum_square


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
